File: app/src/main/java/com/example/android/sd/MainActivity.java
package com.example.android.sd;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.CoordinatorLayout;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import butterknife.BindView;
import butterknife.ButterKnife;
import utils.Variables;

import static utils.FunctionKit.getSnackBar;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.firstPage_CoordinateLayout)
    CoordinatorLayout mCoordinateLayout;
    @BindView(R.id.firstPage_ServiceLoad)
    EditText serviceLoadEditText;
    @BindView(R.id.firstPage_TC_radio)
    RadioGroup RG_TensionCompression;
    @BindView(R.id.firstPage_Tension_Radio)
    RadioGroup RG_TensionOptions;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ButterKnife.bind(this);

        final boolean[] isTension = {true};
        RG_TensionCompression.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup radioGroup, int id) {
                switch (id){
                    case R.id.firstPage_Tension:
                        isTension[0] = true;
                        RG_TensionOptions.setVisibility(View.VISIBLE);
                        break;
                    case R.id.firstPage_Compression:
                        isTension[0] = false;
                        RG_TensionOptions.setVisibility(View.GONE);
                        break;
                }
            }
        });
        FloatingActionButton myFab = (FloatingActionButton) findViewById(R.id.firstPage_FAB);
        myFab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(!TextUtils.isEmpty(serviceLoadEditText.getText())){
                    Intent intent;
                    if (isTension[0]) {
                        if (isBolted()) {
                            intent = new Intent(getBaseContext(), BoltActivity.class);
                        } else {
                            intent = new Intent(getBaseContext(), WeldActivity.class);
                        }
                    } else {
                        intent = new Intent(getBaseContext(), CompressionActivity.class);
                    }
                    intent.putExtra(Variables.serviceLoad, serviceLoadEditText.getText().toString());
                    startActivity(intent);
                } else {
                    Snackbar snackBar = getSnackBar(mCoordinateLayout,R.string.enter_service_load);
                    snackBar.show();
                }
            }
        });
    }

    private boolean isBolted(){
        RadioButton bolt = (RadioButton) findViewById(R.id.firstPage_Bolted);
        return bolt.isChecked();
    }
}

File: app/src/main/java/com/example/android/sd/ui/MainActivity.kt
package com.example.android.sd.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import com.example.android.sd.R
import kotlinx.android.synthetic.main.activity_main.*


/**
 * Created by Kundan on 12-09-2018.
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        init()
    }

    private fun init() {
        initializeRecyclerView()
    }

    private fun initializeRecyclerView() {
        val designTypeList = mutableListOf<DesignType>()
        designTypeList.add(DesignType(R.mipmap.ic_launcher, getString(R.string.steel_design_text)))
        designTypeList.add(DesignType(R.mipmap.ic_launcher, getString(R.string.rcc_design_text)))

        design_type_rv.layoutManager = LinearLayoutManager(this)
        design_type_rv.adapter = DesignTypeAdapter(designTypeList, this)
    }

    class DesignType(var imageID: Int = R.mipmap.ic_launcher, var designType: String = "Design Type")
}